### ISSUE TITLE
clone repository where a given file/directory is maintained

### ISSUE THREAD

**Issue by melissawen** [kworkflow/kworkflow#255]
The maintainership of Linux kernel source code is divided into different subsystem maintainers and repositories, as presented in https://lwn.net/Articles/737093/.

In Linux, scripts/get_maintainers can list the subsystem maintainer tree, in addition to maintainers and mailing lists of a given file/directory. Users get this information via `--scm` option. In the _MAINTAINERS_ file, this information is in each line **T:**, as described in: 
* https://www.kernel.org/doc/html/latest/process/submitting-patches.html#obtain-a-current-source-tree
* https://www.kernel.org/doc/html/latest/process/maintainers.html#descriptions-of-section-entries-and-preferred-order

With this in mind, we can expand the `kw maintainers` feature by allowing users to check and automated clone the subsystem repository related to a file/directory. In other words, using an option on `kw maintainers`, users would specify whenever they want to clone and, if so, where (path) to clone it.

**Comment by rodrigosiqueira** [comment]
Hi @melissawen,

In this case, the user needs to download `get_maintainers.pl` and run the command that clones the project?

I was thinking of making this more flexible by expanding the `init` feature. Follows a CLI draft for this:

1. Show upstream subsystem
```
kw init --list
1. AMD DISPLAY CORE
...
```
Maybe kw can reference this link: https://www.kernel.org/doc/linux/MAINTAINERS

2. `kw init --clone="AMD DISPLAY CORE" [--path=<**path**>]`

What do you think?

**Comment by melissawen** [comment]
yes, doing it in `kw init` makes more sense and I like the --list option (it looks more user-friendly).

Thinking better now:
There is a risk for new users to clone the same repository for different drivers under the same subsystem. So, I wonder if we can have in `kw maintainer` a kind of check repository option to compare if we are developing in the correct one or need to clone another. It will not solve this unsolved issue, but I think it matches the kw maintainers purpose. Does it make sense?
